db.go

```diff
@@ -261,7 +262,7 @@ func scanDocument(iter document.Iterator) (types.Document, error) {
 	}
 
 	if d == nil {
-		return nil, errs.ErrDocumentNotFound
+		return nil, errors.New(errs.ErrDocumentNotFound)
```

Comment:
It's already done as you suggest, but because how it is declared, the stack trace is meaningless. 
We could wrap that in a `NewErrDocFound()` function, possibly. 

I believe that we'd better address those in a different PR, so we can take a step back on how we want to handle the errors, and keep this one simpler.